cmake/scripts/common/ModuleHelpers.cmake

```diff
@@ -48,3 +50,20 @@ function(get_archive_name module_name)
   endif()
 
 endfunction()
+
+# Macro to factor out the repetitive URL setup
+macro(SETUP_BUILD_VARS)
+  get_archive_name(${MODULE_LC})
+  string(TOUPPER ${MODULE_LC} MODULE)
+
+  # allow user to override the download URL with a local tarball
+  # needed for offline build envs
+  if(${MODULE}_URL)
+    get_filename_component(${MODULE}_URL "${${MODULE}_URL}" ABSOLUTE)
+  else()
+    set(${MODULE}_URL ${${MODULE}_BASE_URL}/${ARCHIVE})
```

Comment:
I must be blind, can't find where this is set.. should that be ${${MODULE}_ARCHIVE}?

Reply:
Good spot. fixed